src/Nixfmt/Pretty.hs

```diff
@@ -159,7 +159,10 @@ instance Pretty Binder where
 prettySet :: Bool -> (Maybe Leaf, Leaf, Items Binder, Leaf) -> Doc
 -- Empty attribute set
 prettySet _ (krec, paropen@(LoneAnn _), Items [], parclose@Ann{preTrivia = []}) =
-  pretty (fmap (,hardspace) krec) <> pretty paropen <> hardspace <> pretty parclose
+  pretty (fmap (,hardspace) krec) <> pretty paropen <> sep <> pretty parclose
+  where
+    -- If the braces are on different lines, keep them like that
+    sep = if sourceLine paropen /= sourceLine parclose then hardline else hardspace
```

Comment:
From [CI](https://github.com/NixOS/nixfmt/actions/runs/10003330283/job/27650058187?pr=224) we can see this in the logs for the last commit:

```
[INFO ] #nixfmt-rfc-style: 35101 files processed in 125.91s
```

And from [CI of the previously merged PR](https://github.com/NixOS/nixfmt/actions/runs/9992970991/job/27619275018) we can see this:

```
[INFO ] #nixfmt-rfc-style: 35096 files processed in 132.35s
```

So definitely no significant change there :)

In the future it would be great to include the time in the Nixpkgs diff commit messages